### PR TITLE
Lesson 7 arrays

### PR DESCRIPTION
**Pull request by BTBTravis**
Hey Team here is the lesson seven, array, slides.

### CODE REVIEW

**Comment by BTBTravis** [comment]
still trying to come up with a good class work

File: lessons/7_arrays/index.html [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            <textarea data-template>
+            ### Thinking in arrays
+
+            What about in the user interface of a website?
+
+            - menu items
+            - product tiles
+            - footer links
+
+            Many things can be modled with arrays
```
Comment (MrTim) [inline lessons/7_arrays/index.html]: typo - modeled

File: lessons/7_arrays/index.html [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            <br/>
+            <pre class="fragment"><code data-trim>
+                const names = ["Travis", "Anna", "Ivan", "Carlyon"];
+                const lastPerson = names[names.length - 1];
+                console.log('lastPerson:', lastPerson);
+            </code></pre>
+        </section>
+
+        <section>
+            <h3>remove item from array</h3>
```
Comment (MrTim) [inline lessons/7_arrays/index.html]: All previous section names started from capital letter

**Review by MrTim — COMMENTED** [review]
Nice lesson! 
Suggestions for classwork: create array of favorite book/animals/places/cars/songs and then do some exercises on it like: push new book/place/car/song in your array, then list all odd arrays elements on the page and delete some elements. Maybe we can ask to go to mdn choose a method and try to use it during the class if people would be fast.

**Review by aberezhk — COMMENTED** [review]
I like the content of the lecture. Somehow cant find a way to correctly serve application to get slides working... Please make sure before the lecture that it will work with github pages. As an idea for classwork: input box and button to add to and button to delete by index from array, text field or console output of array content. Also please dont forget to update readme file with links to slides and homeworks

File: lessons/7_arrays/index.html [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            <textarea data-template>
+            # Lecture 7
+            ### Intro to arrays
+            </textarea>
+        </section>
+        <section>
+            <h3>Arrays</h3>
+            <ul>
+                <li>All about storing several pieces of data in one place.</li>
+                <li>They are defined with square brackets and itmes are seprated with commas</li>
```
Comment (tobiastimm) [inline lessons/7_arrays/index.html]:
```suggestion
                <li>They are defined with square brackets and items are separated with commas</li>
```

File: lessons/7_arrays/index.html [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+            <br/>
+            <p>Best to avoid doing:</p>
+            <pre class="fragment"><code data-trim>
+                const myCrazyArray = ["Travis", 10, false, "Carlyon"];
+                console.log('myCrazyArray:', myCrazyArray);
+            </code></pre>
+        </section>
+
+        <section>
+            <h3>Array Acess</h3>
```
Comment (tobiastimm) [inline lessons/7_arrays/index.html]:
```suggestion
            <h3>Accessing Arrays</h3>
```

File: lessons/7_arrays/index.html [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+                }
+                ...
+            </code></pre>
+        </section>
+
+        <section>
+            <h3>What can arrays hold?</h3>
+            <ul>
+                <li>technically they can hold elements of any type including functions</li>
+                <li>but it is a good idea to keep them all the same type because of loops we'll learn about next lesson</li>
```
Comment (tobiastimm) [inline lessons/7_arrays/index.html]:
```suggestion
                <li>you should always try to store data of the same type in an array, especially in regards to looping</li>
                <li>More about loops in the next lesson</li>
```

File: lessons/7_arrays/index.html [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+                const secondName = names[1];
+                console.log('secondName:', secondName);
+            </code></pre>
+        </section>
+
+        <section>
+            <h3>Updating Array Elements</h3>
+            <ul>
+                <li>now that we know how to access array items we can also change them</li>
+                <li>this way of changing already created varable is call mutation</li>
```
Comment (tobiastimm) [inline lessons/7_arrays/index.html]:
```suggestion
                <li>this way of changing an already created variable is called `mutation`</li>
```

File: lessons/7_arrays/index.html [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+                const names = ["Travis", "Anna", "Ivan", "Carlyon"];
+                names[0] = "Bob";
+                console.log('names:', names);
+            </code></pre>
+        </section>
+
+        <section>
+            <h3>Adding to an Array</h3>
+            <ul>
+                <li>arrays can be added to via their push method</li>
```
Comment (tobiastimm) [inline lessons/7_arrays/index.html]:
```suggestion
                <li>elements can be added to an array via the `.push` function</li>
```

File: lessons/7_arrays/index.html [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+                console.log('names pre push:', names);
+                names.push("Tim");
+                console.log('name post push:', names);
+            </code></pre>
+        </section>
+
+        <section>
+            <h3>The length property</h3>
+            <ul>
+                <li>arrays have a very useful perperty `length`</li>
```
Comment (tobiastimm) [inline lessons/7_arrays/index.html]:
```suggestion
                <li>arrays have a very useful property `length`</li>
```

File: lessons/7_arrays/index.html [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+                <li>for example we could use it to get the last item of the array</li>
```
Comment (tobiastimm) [inline lessons/7_arrays/index.html]:
```suggestion
                <li>for example, we could use it to get the last item of an array</li>
```

**Review by tobiastimm — CHANGES_REQUESTED** [review]
Look's right, with typos here and there

**Review by MrTim — APPROVED** [review]

**Pull request merged**